src/client/mod.rs

```diff
@@ -478,23 +480,113 @@ header! {
     (XVaultWrapTTL, "X-Vault-Wrap-TTL") => [String]
 }
 
+#[derive(Deserialize)]
+#[allow(dead_code)]         // We need only one of the key fields
```

Comment:
I'm not following why we need this allow here.

Also, if we only need `root_token`, I believe we don't need to add the other fields, serde will skip unknown fields.